File: JavaScript/7js 1/calculate_total.js
/*
|--------------------------------------------------------------------------
| Calculate Meal Total
|--------------------------------------------------------------------------
|
*/

function calculate_total(subtotal, tax, tip) {
   return Number((subtotal + (subtotal * (tax / 100)) + (subtotal * (tip / 100))).toFixed(2));
}


const calculate_total = (subtotal, tax, tip) => Math.round(subtotal * (100 + tax + tip)) / 100


const calculate_total = (sub, tax, tip) => Math.round(sub * (100 + tax + tip)) / 1e2


function calculate_total(subtotal, tax, tip) {
   return +(subtotal * (1 + tip / 100 + tax / 100)).toFixed(2);
}


console.log(calculate_total(5, 5, 10), 5.75);
console.log(calculate_total(36.97, 7, 15), 45.10);
console.log(calculate_total(0.00, 6, 18), 0.00);
console.log(calculate_total(80.94, 0, 20), 97.13);
console.log(calculate_total(54.96, 8, 0), 59.36);

File: Python/7py 1/calculate_total.py
"""
|--------------------------------------------------------------------------
| Calculate Meal Total
|--------------------------------------------------------------------------
|
"""

def calculate_total(subtotal, tax, tip):
   return float(format(subtotal + (subtotal * (tax / 100)) + (subtotal * (tip / 100)), ".2f"))


def calculate_total(subtotal, tax, tip):
   return round((1 + (tip + tax) / 100.0) * subtotal, 2)


def calculate_total(subtotal, tax, tip):
   return float("{0:.2f}".format(subtotal + subtotal*(tax+tip)/100))


def calculate_total(subtotal, tax, tip):
   return round(subtotal * ( 1 + tax / 100.0 + tip /100.0), 2)


print(calculate_total(5, 5, 10), 5.75)
print(calculate_total(36.97, 7, 15), 45.10)
print(calculate_total(0.00, 6, 18), 0.00)
print(calculate_total(80.94, 0, 20), 97.13)
print(calculate_total(54.96, 8, 0), 59.36)